src/systemctl/systemctl-start-unit.c

```diff
@@ -399,3 +400,41 @@ int verb_start(int argc, char *argv[], void *userdata) {
 
         return ret;
 }
+
+int verb_toggle(int argc, char *argv[], void *userdata) {
+        _cleanup_strv_free_ char **names = NULL;
+        UnitActiveState active_state;
+        sd_bus *bus;
+        int r;
+        bool is_active = false;
+        char *new_args[3];
+
+        if (!argv[1])
+                return 0;
+
+        r = mangle_names("to toggle", strv_skip(argv, 1), &names);
+        if (r < 0)
+                return r;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, names) {
+                is_active = false;
+
+                r = get_state_one_unit(bus, *name, &active_state);
+                if (r < 0)
+                        return r;
+
+                is_active = IN_SET(active_state, UNIT_ACTIVE, UNIT_RELOADING);
+
+                new_args[0] = (char*) (is_active ? "stop" : "start");
+                new_args[1] = basename(*name);
+                new_args[2] = NULL;
+
+                r = verb_start(2, new_args, userdata);
```

Comment:
return value will be overwritten if there's more than one unit, it should be checked immediately

Reply:
Shouldn't this rather try to toggle the remaining units and return the first error then? How do the other verbs behave?

Reply:
I only checked start verb, and it does go through all the units even if there's one erroneous unit among them and return that error. I'll make a fix.

Reply:
Thinking further, maybe it's better to leave it as before without checking? That way it still prints the error message but also continues on toggling other units.

Reply:
The usual pattern we use is something like:

```
int r = 0, k;

k = foo()
if (k < 0 && r == 0)
     r = k;
```

This way if the first one errors out and the last one doesn't, we still return an error, which is important as that's what's expected